### ISSUE TITLE
Danger signs questions not appearing for YI

### ISSUE THREAD

**Issue by kaledip** [WorldHealthOrganization/smart-emcare-cmr#18]

**Comment by kaledip** [comment]
"Obstructed or Absent Breathing" and "Convulsing now" need to be shown for YI

**Comment by magp18** [comment]
Hi Dipti, danger signs is not asked for YI, I think this was agreed on previously so it's not defined in the L4 as part of the YI workflow, but we can either : 

- Allow danger signs to show for YI show it with conditional logic for the different ages inside
- Create a danger signs questionnaire for YI 
- Move the questions to a different questionnaire

**Comment by kaledip** [comment]
Earlier for Iraq there were no danger signs for YI but for Cameroon we do have specific questions. I think you can take up any suitable approach for the same.

**Comment by magp18** [comment]
What do you think @ppanchal-argus ? I think it makes sense to have it in the danger signs and constraint it in the condition with the age since it is danger signs. But if we do this we would have to update the IRAQ because some elements don't have age related constraints assuming it won't show for YI, right @delcroip

**Comment by ppanchal-argus** [comment]
The approach of constraining the condition with age LGTM! 
Also isn't it possible to just update the cameroon questionnaire?

**Comment by magp18** [comment]
Yes @ppanchal-argus but (correct me if I am wrong cause I am not sure 😅 ) as plan definition is not working yet we are gluing the questionnaires on L4 level but if I am not mistaken if it's a yi (<2months) the questionnaire about danger signs is not showing so even if there are data elements for yi, the questionnaire will not load. 
`val consultationFlowStageListUnderTwoMonths = arrayListOf<String?>(
    CONSULTATION_STAGE_REGISTRATION_PATIENT,
    CONSULTATION_STAGE_REGISTRATION_ENCOUNTER,
    CONSULTATION_STAGE_MEASUREMENTS,
    CONSULTATION_STAGE_SYMPTOMS,
    CONSULTATION_STAGE_SIGNS,
    CONSULTATION_STAGE_ASSESSMENTS,
    CONSULTATION_STAGE_CLASSIFICATIONS,
    CONSULTATION_STAGE_TREATMENTS
)`

vs older than 2 monts: 
`val consultationFlowStageList = arrayListOf<String?>(
    CONSULTATION_STAGE_REGISTRATION_PATIENT,
    CONSULTATION_STAGE_REGISTRATION_ENCOUNTER,
    CONSULTATION_STAGE_DANGER_SIGNS,
    CONSULTATION_STAGE_MEASUREMENTS,
    CONSULTATION_STAGE_SYMPTOMS,
    CONSULTATION_STAGE_SIGNS,
    CONSULTATION_STAGE_ASSESSMENTS,
    CONSULTATION_STAGE_CLASSIFICATIONS,
    CONSULTATION_STAGE_TREATMENTS
)`

**Comment by ppanchal-argus** [comment]
Yes you are totally right! @magp18 
If we go forward with integrating the danger signs questionnaire in cameroon for YI, we may have to add logic to show danger signs based on country.
Moreover, this workaround will only be temporary since it won't be required once plan-definition runs successfully!
